gsi/gssapi/source/library/init_sec_context.c

```diff
@@ -202,6 +202,13 @@ GSS_CALLCONV gss_init_sec_context(
             *ret_flags = 0 ;
         }
     }
+    else if (context->gss_state == GSS_CON_ST_TOKEN1 &&
+             input_token->length == 0) {
+        /*
+         * This is for compatibility with the dcache server implementation
+         */
+        context->gss_state = GSS_CON_ST_FLAGS;
+    }
```

Comment:
@ellert:
IIUC this forwards the connection state (as defined in https://github.com/gridcf/gct/blob/f8ccd9058e9d54d0cc9430818b0a4a9bd65b9b1d/gsi/gssapi/source/library/globus_gsi_gss_constants.h#L94..L106) for the client by skipping the `GSS_CON_ST_TOKEN2` state if no or an empty input token follows the output token that follows the first input token. This enables the GCT against dCache case when TLS v1.3 is used but doesn't harm the GCT against GCT case for TLS v1.3, right?

Why not include the skipping below https://github.com/gridcf/gct/blob/5a239381ad4444a8e96c9d2154251586f55f48f9/gsi/gssapi/source/library/init_sec_context.c#L421 ...?

Reply:
I have rewritten the code according to your suggestion.

Reply:
Thanks.